### ISSUE TITLE
New Python terminal input is sometimes a bit sluggish

### ISSUE THREAD

**Issue by tjclement** [hackzone/cz20-badge#21]
Perhaps due to the frequency with which we poll the ESP for new output?

**Comment by jorisplusplus** [comment]
Poll rate is 500ms which is causing the sluggish feel since we dont show the typed letters only the letters written by micropython. Increasing the polling rate will increase overhead and increase chance of the webusb stack crashing

**Comment by tjclement** [comment]
Yes, so maybe we can consider typing offline into the terminal and flushing on return/tab. That of course increases the complexity of our code, but it makes the user experience much nicer imo
